addons/website/static/src/snippets/s_dynamic_filter_carousel/000.js

```diff
@@ -0,0 +1,122 @@
+odoo.define('website.s_dynamic_filter_carousel', function (require) {
+'use strict';
+
+const config = require('web.config');
+const core = require('web.core');
+const publicWidget = require('web.public.widget');
+
+const qweb = core.qweb;
+
+const DynamicFilterCarousel = publicWidget.Widget.extend({
+    selector: '.s_dynamic_filter_carousel',
+    xmlDependencies: ['/website/static/src/xml/website.s_dynamic_filter_carousel.xml'],
+    disabledInEditableMode: false,
+
+    /**
+     * @override
+     */
+    init: function () {
+        this._super.apply(this, arguments);
+        this.uniqueId = _.uniqueId('s_dynamic_filter_carousel_');
+        this._onResizeChange = _.debounce(this._refresh_carousel_content, 100);
+    },
+    /**
+     * @override
+     */
+    start: function () {
+        return this._super.apply(this, arguments)
+            .then(() => {
+                $(window).resize(() => {
+                    this._onResizeChange();
+                });
+                this._refresh_carousel_content();
+            });
+    },
+    /**
+     * @override
+     */
+    destroy: function () {
+        this._super(...arguments);
+        this.$el.addClass('d-none');
+        this.$el.find('.carousel-inner').html('');
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    _create_dataGroups: function (data) {
```

Comment:
seems complicated to read to make something simple at first view :/

```js
    _create_dataGroups: function (data) {
        const data_groups = [];
        const records_per_chunk = this._number_of_records_per_chunk();
        const number_of_slide = Math.ceil(data.length / records_per_chunk);
        for (const i of Array(number_of_slide).keys()) {
            let one_group = [];
            for (const item of data.slice(i * records_per_chunk, (i + 1) * records_per_chunk)) {
                one_group.push(
                    qweb.render(
                        'website.s_dynamic_filter_carousel.one_data_group',
                       { 'item': item, 'class': 'col-md-' + Math.floor(12/number_of_records_per_chunk)}
                    )
                );
            }
            data_groups.push(one_group)
        };

```

```xml
    <t t-name="website.s_dynamic_filter_carousel.one_data_group">
        <div t-att-class='class'>
            <div class='card h-100'>
                <t t-raw="item"/>
            </div>
        </div>
    </t>
```
Ok at this end, it is more readable maybe but still complicated for not a big deal.
Final question, is why do we want group of group at a moment ? we don't to anything with it...

So we could just have 
```js
        for (const i of Array(number_of_slide).keys()) {
            data_groups.push(
                qweb.render(
                    'website.s_dynamic_filter_carousel.one_data_group',
                   {
                       'items': data.slice(i * records_per_chunk, (i + 1) * records_per_chunk),
                       'class': 'col-md-' + Math.floor(12/number_of_records_per_chunk)
 }
                )
            );
        };
```

```xml
    <t t-name="website.s_dynamic_filter_carousel.one_data_group">
        <div t-att-class='class' t-foreach="items" t-as="item">
            <div class='card h-100'>
                <t t-raw="item"/>
            </div>
        </div>
    </t>
```

But final question is why do we want group at a moment: and not just drop the method _create_dataGroups and do the split in website.s_dynamic_filter_carousel.carousel template directly